src/Bonsai.ML.HiddenMarkovModels/Observations/AutoRegressiveObservations.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using System;
+
+namespace Bonsai.ML.HiddenMarkovModels.Observations
+{
+    /// <summary>
+    /// Represents an autoregressive observations model.
+    /// </summary>
+    [JsonObject(MemberSerialization.OptIn)]
+    public class AutoRegressiveObservations : ObservationsModel
+    {
+
+        private int lags = 1;
+
+        /// <summary>
+        /// The lags of the observations for each state.
+        /// </summary>
+        public int Lags { get => lags; set {lags = value; UpdateString(); } }
+
+        /// <summary>
+        /// The As of the observations for each state.
+        /// </summary>
+        public double[,,] As { get; private set; } = null;
```

Comment:
I tried sticking as closely as possible to the original API defined in the [ssm package](https://github.com/lindermanlab/ssm/tree/master/ssm). They tend to keep the "s"s throughout (for example, `observations.py` and [`Observations`](https://github.com/lindermanlab/ssm/blob/6c856ad3967941d176eb348bcd490cfaaa08ba60/ssm/observations.py#L18C1-L18C28) instead of `Observation`). In the same way, properties of classes also tend to keep the "s"s (example [`As`](https://github.com/lindermanlab/ssm/blob/6c856ad3967941d176eb348bcd490cfaaa08ba60/ssm/observations.py#L848) inside of `AutoRegressiveObservations`). I'm more inclined to stick with this naming convention to maintain consistency between the 2 APIs, but I'm open to changing this.